packages/config/src/utils/extensions/utils.ts

```diff
@@ -0,0 +1,242 @@
+import { NetlifyAPI } from 'netlify'
+
+import { JIGSAW_URL } from './constants.js'
+
+export type Extension = {
+  id: string
+  name: string
+  slug: string
+  hostSiteUrl: string
+  installedOnTeam: boolean
+}
+
+export const getInstalledExtensionsForSite = async ({
+  accountId,
+  siteId,
+  netlifyToken,
+}: {
+  accountId: string
+  siteId: string
+  netlifyToken: string
+}) => {
+  const url = new URL(
+    `${JIGSAW_URL}/team/${encodeURIComponent(accountId)}/integrations/installations/${encodeURIComponent(siteId)}`,
+    JIGSAW_URL,
+  )
+  const extensionsResponse = await fetch(url.toString(), {
+    headers: {
+      'netlify-token': netlifyToken,
+      'Api-Version': '2',
+    },
+  })
+  if (!extensionsResponse.ok) {
+    throw new Error(`Failed to fetch extensions`)
+  }
+
+  const extensions = (await extensionsResponse.json()) as {
+    id: number
+    name: string
+    integrationId: number
+    integrationSlug: string
+  }[]
+
+  return extensions
+}
+
+export type InstallExtensionResult =
+  | {
+      slug: string
+      error: null
+    }
+  | {
+      slug: string
+      error: {
+        code: string
+        message: string
+      }
+    }
+
+export const installExtension = async ({
+  netlifyToken,
+  accountId,
+  slug,
+  hostSiteUrl,
+}: {
+  netlifyToken: string
+  accountId: string
+  slug: string
+  hostSiteUrl: string
+}): Promise<InstallExtensionResult> => {
+  const { data: jigsawToken, error } = await getJigsawToken({
+    netlifyToken: netlifyToken,
+    accountId,
+    integrationSlug: slug,
+    isEnable: true,
+  })
+  if (error || !jigsawToken) {
+    return {
+      slug,
+      error: {
+        code: 'FAILED_TO_GET_JIGSAW_TOKEN',
+        message: error?.message ?? 'Unknown error',
+      },
+    }
+  }
+
+  const extensionOnInstallUrl = new URL('/.netlify/functions/handler/on-install', hostSiteUrl)
+  const installedResponse = await fetch(extensionOnInstallUrl, {
+    method: 'POST',
+    body: JSON.stringify({
+      teamId: accountId,
+    }),
+    headers: {
+      'netlify-token': jigsawToken,
+    },
+  })
+
+  if (!installedResponse.ok && installedResponse.status !== 409) {
+    const text = await installedResponse.text()
+    return {
+      slug,
+      error: {
+        code: 'FAILED_TO_INSTALL_EXTENSION',
+        message: text,
+      },
+    }
+  }
+  return {
+    slug,
+    error: null,
+  }
+}
+
+type JigsawTokenResult =
+  | {
+      data: string
+      error: null
+    }
+  | {
+      data: null
+      error: { code: number; message: string }
+    }
+
+const getJigsawToken = async ({
```

Comment:
I think you should be able to do this without getting the jigsaw token first. Try it please, I don't want to use the react-ui to do this work here.

The buildbot token should already be a JWT I believe. 
The CLI token is a user token and should work without first becoming a jigsaw token _I think_. Worth trying.

Reply:
It works! removed `getJigsawToken` now.